libraries/Update/src/OTA_Update.h

```diff
@@ -0,0 +1,10 @@
+#include "esp_http_client.h"
+#define HttpEvent_t esp_http_client_event_t
+class OTA_UpdateClass {
+
+    public:
+    void start_firmware_upgrade(const char * cert, const char * url, bool skip_cert_cn_check);
+    void onHttpEvent(void (*ptr)(HttpEvent_t *));
+};
+
+extern OTA_UpdateClass OTA;
```

Comment:
use AWS_OTA and not OTA. You are reserving a high level namespace for just one specific implementation. If AWS is not fitting, maybe HTTPS or something. not just OTA. this goes for the file names, class names and the exposed object.